### PR TITLE
Fixes #46 - isc dhcp configuration check

### LINKED ISSUE #46: Procedure: isc dhcp configuration check
Manual edits in ISC DHCP often lead to "Cannot add Subnet" error or other parsing issues as our parser is very limited. This procedure (should not be really check but a "command" you can run when needed) would do this (tested on Satellite 6.1+):

```shell
#!/bin/bash
curl -ks --cert /etc/foreman/client_cert.pem --key /etc/foreman/client_key.pem --cacert /etc/foreman/proxy_ca.pem https://$(hostname):9090/dhcp
dhcpd -t -cf /etc/dhcp/dhcpd.conf
```

The first command returns JSON of subnet data, error message otherwise. The second is syntax check of ISC, returns some text (always), we need check the return value instead.

### PR DESCRIPTION
**Pull request by kgaikwad**

### CODE REVIEW

File: lib/foreman_maintain/concerns/system_helpers.rb [hunk 1]
```diff
@@ -92,6 +92,12 @@ def execute(command, options = {})
       def shellescape(string)
         Shellwords.escape(string)
       end
+
+      def json_parse(json_string)
+        JSON.parse(json_string)
+      rescue
+        nil
```
Comment (swapab) [inline lib/foreman_maintain/concerns/system_helpers.rb]: You can rescue with `{}` or `''`

File: definitions/features/foreman_proxy.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
+class Features::ForemanProxy < ForemanMaintain::Feature
+  metadata do
+    label :foreman_proxy
+  end
+
+  class DhcpCurlResponse
+    attr_reader :result, :http_code, :error_msg
+
+    def initialize(response, http_code, http_error_msg)
+      @result = response || ''
```
Comment (swapab) [inline definitions/features/foreman_proxy.rb]: Memoization can be optional if `json_parse` returns `{}` or `''`

File: definitions/features/foreman_proxy.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    http_line.split(curl_http_status.to_s).last.strip
+  end
+
+  def run_dhcp_curl
+    curl_resp = execute(dhcp_curl_cmd)
+    array_output = curl_resp.split(/\r\n/)
+    result_array = array_output.last.split(/\n/)
+    curl_http_status = result_array.delete_at(result_array.length - 1).strip.to_i
+    curl_http_resp = json_parse(result_array.join(''))
+    DhcpCurlResponse.new(
```
Comment (kgaikwad) [inline definitions/features/foreman_proxy.rb]: @swapab ,
I am using this method `run_dhcp_curl` within this feature file only.
Please let me know if you want me to change the logic here.

Comment (swapab) [inline definitions/features/foreman_proxy.rb]: Logic is okay. Just ought to keep features clean and to its purpose.
`DhcpCurlResponse` can be `Utils::Dhcp`

**Comment by kgaikwad** [comment]
@iNecas, @swapab 
Resolved conflicts. Now, it is ready for review.
If no any additional changes required then can we merge this branch please?
For a [dhcp issue](https://github.com/theforeman/foreman_maintain/issues/75), I can use feature `foreman_proxy`.

**Comment by swapab** [comment]
LGTM :+1:

**Comment by kgaikwad** [comment]
To test this PR, you will need to enable DHCP service on your satellite instance.

By default the integrated capsule server have Pulp, Puppet, Puppet CA feature enabled.
To enable DNS, DHCP, TFTP for inbuilt capsule server on the Red Hat Satellite v 6.x server execute command below:

- Satellite v 6.1
`katello-installer --capsule-dhcp true --capsule-dhcp-interface eth0 --capsule-dhcp-range "xx.xx.xx.xx xx.xx.xx.xx" --capsule-dhcp-gateway xx.xx.xx.xx --capsule-dhcp-nameservers xx.xx.xx.xx  --capsule-dns true --capsule-dns-interface eth0 --capsule-dns-forwarders xx.xx.xx.xx --capsule-dns-reverse xx.xx.xx.in-addr.arpa --capsule-dns-server xx.xx.xx.xx --capsule-tftp true --capsule-tftp-servername $(hostname -f) `

- Satellite v 6.2
`satellite-installer --foreman-proxy-dhcp true --foreman-proxy-dhcp-gateway  xx.xx.xx.xx --foreman-proxy-dhcp-interface "eth1" --foreman-proxy-dhcp-nameservers  xx.xx.xx.xx --foreman-proxy-dhcp-range " xx.xx.xx.xx  xx.xx.xx.xx" --foreman-proxy-dhcp-server  xx.xx.xx.xx --foreman-proxy-dns true --foreman-proxy-dns-forwarders  xx.xx.xx.xx --foreman-proxy-dns-interface "eth1" --foreman-proxy-dns-reverse " xx.xx.xx.in-addr.arpa" --foreman-proxy-dns-server  xx.xx.xx.xx --foreman-proxy-tftp true --foreman-proxy-tftp-servername $(hostname -f)`

Then verify below things -
1. `dhcpd` service is running on sat instance
2. Check if `/etc/dhcp/dhcpd.conf` exists

To test this check using foreman_maintain, try to edit `/etc/dhcp/dhcpd.conf` file and add some wrong syntax changes to a file. This check will give you warning message for incorrect syntax.

**Pull request merged**